media_driver/linux/common/os/mos_gpucontext_specific.cpp

```diff
@@ -112,6 +115,123 @@ MOS_STATUS GpuContextSpecific::Init(OsContext *osContext)
 
     m_GPUStatusTag = 1;
 
+    m_createOptionEnhanced = (MOS_GPUCTX_CREATOPTIONS_ENHANCED*)MOS_AllocAndZeroMemory(sizeof(MOS_GPUCTX_CREATOPTIONS_ENHANCED));
+    m_createOptionEnhanced->SSEUValue = createOption->SSEUValue;
+
+    if (typeid(*createOption) == typeid(MOS_GPUCTX_CREATOPTIONS_ENHANCED))
+    {
+        PMOS_GPUCTX_CREATOPTIONS_ENHANCED createOptionEnhanced = static_cast<PMOS_GPUCTX_CREATOPTIONS_ENHANCED>(createOption);
+        m_createOptionEnhanced->UsingSFC = createOptionEnhanced->UsingSFC;
+    }
+
+    if (osInterface->ctxBasedScheduling)
+    {
+        m_i915Context = mos_gem_context_create_shared(osInterface->pOsContext->bufmgr,
+                                             osInterface->pOsContext->intel_context,
+                                             I915_CONTEXT_CREATE_FLAGS_SINGLE_TIMELINE);
+        if (m_i915Context == nullptr)
+        {
+            MOS_OS_ASSERTMESSAGE("Failed to create context.\n");
+            return MOS_STATUS_UNKNOWN;
+        }
+        m_i915Context->pOsContext = osInterface->pOsContext;
+
+        m_i915ExecFlag = I915_EXEC_DEFAULT;
+        if (GpuNode == MOS_GPU_NODE_3D || GpuNode == MOS_GPU_NODE_COMPUTE)
+        {
+            struct i915_engine_class_instance engine_map;
+            engine_map.engine_class = I915_ENGINE_CLASS_RENDER;
+            engine_map.engine_instance = 0;
+
+            if (mos_set_context_param_load_balance(m_i915Context,&engine_map, 1))
+            {
+                MOS_OS_ASSERTMESSAGE("Failed to set balancer extension.\n");
+                return MOS_STATUS_UNKNOWN;
+            }
+
+            if (createOption->SSEUValue != 0)
+            {
+                struct drm_i915_gem_context_param_sseu sseu;
+                MOS_ZeroMemory(&sseu, sizeof(sseu));
+                sseu.engine.engine_class = I915_ENGINE_CLASS_RENDER;
+                sseu.engine.engine_instance = 0;
```

Comment:
Would it be more self-documenting to replace 0 with m_i915ExecFlag? I am thinking how to designate the VENG slot.